### DISCUSSION TITLE
Error | appui-react.ContentGroup | getControlFromElement: no control found for element

### DISCUSSION THREAD

**Discussion by Snailya** [iTwin/itwinjs-core#3540] [Q&A]
I create  a frontstage with custom control as content view. That content view couldn't load as expected by when calling FrontsageManager.setActiveFrontstage, and an error throws: no control found for element.
What does this error point to?
I also found that when I comment out the setActiveFronstage during the app is running, the content displays.

**Comment by bsteinbk** [comment]
@Snailya - Typically this error would occur if Id's where not specified in the ContentGroupProps or the ContentProps.  Below is an example where the ContentGroup id is set to `"ui2-frontstage-main-content-group"` while the ContentProps id is set to `"primaryContent"`.

```
    return new ContentGroup({
      id: "ui2-frontstage-main-content-group",
      layout: StandardContentLayouts.singleView,
      contents: [
        {
          id: "primaryContent",
          classId: IModelViewportControl.id,
          applicationData: {
            supplyViewOverlay: FrontstageUi2ContentGroupProvider.supplyViewOverlay,
            isPrimaryView: true,
            supports: ["issueResolutionMarkers", "viewIdSelection", "3dModels", "2dModels"],
            viewState: UiFramework.getDefaultViewState,
            iModelConnection: UiFramework.getIModelConnection,
          },
        },
      ],
    });
```
Without seeing  the code for your application, I can't say why the call to setActiveFrontstage can be commented out yet the frontstage content is still being displayed.

**Reply by Snailya** [reply]
Here's my code.
![image](https://user-images.githubusercontent.com/19658624/164578075-6c2e96d9-786d-403f-8672-2b9e9f169265.png)
![image](https://user-images.githubusercontent.com/19658624/164578085-d79c3c81-8c27-4de6-8af9-578ac285844a.png)
![image](https://user-images.githubusercontent.com/19658624/164578198-c195f5f2-cb69-4034-becc-b2ec09954723.png)

**Reply by Snailya** [reply]
After steping into the code, now I think this is irrelevant to  no control found for element error, as this error raises when ContentViewManager trying to get the old content which is nothing on first startup.
However, why the content not show remains unknown.
